convert2rhel/pkghandler.py

```diff
@@ -1101,52 +1068,171 @@ def clean_yum_metadata():
 
-        pkg_1 = 'kernel-core-0:4.18.0-240.10.1.el8_3.x86_64'
-        pkg_2 = 'kernel-core-0:4.18.0-239.0.0.el8_3.x86_64'
+    :param version1: The version to be compared.
+    :type version1: str
+    :param version2: The version to compare against.
+    :type version2: str
+
+    .. example::
 
-    The output of this will be a tuple containing the package version in a
-    tuple::
+        >>> match = compare_package_versions("kernel-core-5.14.10-300.fc35", "kernel-core-5.14.15-300.fc35")
+        >>> print(match) # -1
 
-        result = _package_version_cmp(pkg_1, pkg_2)
-        print("Result is: %s" % result)
-        # Result is: -1
+    .. note::
 
-    The function will ignore the package name as it is not an important
-    information here and will only care about the version that is tied to it's
-    name.
+        Since the return type is a int, this could be difficult to understand
+        the meaning of each number, so here is a list that represents
+        every possible number:
+            * -1 if the version1 is less then version2 version
+            * 0 if the version1 is equal version2 version
+            * 1 if the version1 is greater than version2 version
 
-    :param pkg_1: The first package to extract the version
-    :type pkg_1: str
-    :param pkg_2: The second package to extract the version
-    :type pkg_2: str
-    :return: An integer resulting in the package comparision
+    :return: Return a number indicating if the versions match, are less or greater then.
     :rtype: int
     """
+    # call parse_pkg_string to obtain a list containing name, epoch, release, version and arch fields
+    # order of fields returned: name, epoch, version, release, arch
+    version1_components = parse_pkg_string(version1)
+    version2_components = parse_pkg_string(version2)
+
+    # ensure package names match, error if not
+    if version1_components[0] != version2_components[0]:
+        raise ValueError("%s does not match %s" % (version1_components[0], version2_components[0]))
+
+    # ensure package arches match, error if not
+    if version1_components[4] != version2_components[4] and all(([version1_components[4]], version2_components[4])):
+        raise ValueError("%s does not match %s" % (version1_components[4], version2_components[4]))
+
+    # create list containing EVR for comparison
+    evr1 = (version1_components[1], version1_components[2], version1_components[3])
+    evr2 = (version2_components[1], version2_components[2], version2_components[3])
+
+    return rpm.labelCompare(evr1, evr2)
+
+
+def parse_pkg_string(pkg):
+    """
+    This function takes a version string in NEVRA, NEVR, NVRA, NVR, ENVRA, ENVR and decides whether
+    to parse with a yum/dnf module based on the package manager type of the system.
+
+    :param pkg: The package to be parsed.
+    :type pkg: str
+
+    :return: Return a Return a list containing name, epoch, version, release, arch
+    :rtype: list[str | None]
```

Comment:
I think I did the wrong suggestion here about the return type. The deal with this one that I suggested is that it will be either a `['something']` or `[]`. 

```suggestion
    :rtype: list[str]
```

Reply:
I thought it looked correct, the fields in the list will either be None type or a string

Reply:
After some discussion, it turns out lists can only accept one type. Adding this suggestion